### ISSUE TITLE
RetinaNet doesn't evaluate on TPUStrategy

### ISSUE THREAD

**Issue by tanzhenyu** [keras-team/keras-cv#1077]
When training with:
`model.fit(train_ds, validation_data=eval_ds)`

adding padding function would work around the Ragged Tensor issue, but evaluation still fails with:
2022-12-01 22:11:05.662365: E tensorflow/core/tpu/kernels/tpu_compilation_cache_external.cc:113] Detected unsupported operations when trying to compile graph cluster_test_function_671004466589259261[] on XLA_TPU_JIT: CombinedNonMaxSuppression (No registered 'CombinedNonMaxSuppression' OpKernel for XLA_TPU_JIT devices compatible with node {{node nms_prediction_decoder/non_max_suppression/combined_non_max_suppression/CombinedNonMaxSuppression}}){{node nms_prediction_decoder/non_max_suppression/combined_non_max_suppression/CombinedNonMaxSuppression}}
One approach is to outside compile the unsupported ops to run on CPUs by enabling soft placement `tf.config.set_soft_device_placement(True)`. This has a potential performance penalty.

This is unfortunately a TPU issue by itself, currently this is being worked around by overriding `model.predict`, but seems `model.evaluate` is still failing

**Comment by bhack** [comment]
Have you try to use the new bridge?

**Comment by bhack** [comment]
Check my previous gist at https://github.com/keras-team/keras-cv/pull/1021#issuecomment-1317925675

**Comment by tanzhenyu** [comment]
> #1021 (comment)

I think the function itself runs fine w/ or w/o the new bridge. It's the strategy.run that poses the problem (still, both w/ or w/o the bridge)

**Comment by bhack** [comment]
I was talking about your last error `CombinedNonMaxSuppression (No registered 'CombinedNonMaxSuppression' OpKernel for XLA_TPU_JIT` after the padding workaround

**Comment by tanzhenyu** [comment]
Fixed due to recent refactoring
